### ISSUE TITLE
Ricing Information for Newbie [Not an Issue]

### ISSUE THREAD

**Issue by AnoRebel** [worron/awesome-config#53]
So, I really like your setup and would like to clone it and RICE it, but Lua isn't in my language knowledge, but for this setup, am willing to learn some..
I checked other setups like the glorious-dotfiles but it's abit much for me now, this is the perfect setup.
So I would like to ask some few(hopefully) questions about your setup, if you wouldn't mind..

1. Folder Structure, how is your code structured. Eg. What each folder is based on.
2. Code Structure, how is the code organised. Eg. I want to remove the desktop widgets, but the code is highly (should i say) 'inherited', i dont know where to begin.
3. What components are distro specific, because I'm currently on Mint and I noticed you're on Arch?
4. Any caveats to know about while Ricing?

Basically, I'm just asking for alittle help to start Ricing this amazing setup, if it wont be a bother.

Consider creating a tutorial or wiki for this, its honestly amazing.

**Comment by worron** [comment]
>Folder Structure, how is your code structured. Eg. What each folder is based on.

>Code Structure, how is the code organised. Eg. I want to remove the desktop widgets, but the code is highly (should i say) 'inherited', i dont know where to begin.

First of all `redflat` submodule consist of core mechanics scripts. Better not to touch anything under that folder during first steps.

Main config files are under  `colorless`, `color`, `shade`, еvery sub-folder  there is a themed config (I associate them with colors). Those files define general setup structure (panel widgets, desktop widgets, hotkeys, windows behavior). Every colored config has a root file `rc-*.lua`, start with it. For example to disable desktop widgets in blue themed config you need [this block](https://github.com/worron/awesome-config/blob/54e545a70df9c17d4c95fcb7c16edff6173e2e20/color/blue/rc-blue.lua#L284-L292). Also some files may be cross linked between colored configs,  just pay attention to every `require` you'll met.

Almost all visual appearance (colors, fonts, widget sizes) configuration are split from 'structure' files and can found under `themes` folder. Sub-folders there also associated with configs by colors.

>What components are distro specific, because I'm currently on Mint and I noticed you're on Arch?

I pointed the most environment specific parts in `Readme` 'Start with Colors' section.

>Any caveats to know about while Ricing?

No caveats in my mind, but some tips. Put Awesome API documentation and any lua tutorial to your bookmarks from the beginning. Think of  this and others configs as a merely examples, do not hesitate to change and adapt, Awesome WM is disigned for it. Have fun.

**Comment by AnoRebel** [comment]
Thanks a lot, this is quite helpful.
Am gonna close this for now and get started but I'll open it again if I need more guidance.
Again, thanks.
And great work!

**Comment by AnoRebel** [comment]
I've started to  slowly understand the file system but i haven't found where to setup the startup applications??

**Comment by worron** [comment]
>I've started to slowly understand the file system but i haven't found where to setup the startup applications?

[autostart-config.lua](https://github.com/worron/awesome-config/blob/master/color/blue/autostart-config.lua)? Also consider using any of WM irrelevant methods to applications autostart in Linux.

>And whats the keybinding to show all the set Key-binds?? like the default +s??

`Mod4+F1`?

**Comment by AnoRebel** [comment]
Cool, thanks again for the fast reply.. Okay, I'll check out other methods to, was just asking also to interfere with any configured methods you had to avoid overwriting each other..
Aah, found the Keybindings key, thanks still..
Okay, onto more hacking and Ricing, will update you in case i get stuck again..
Thanks again.
